src/components/Input.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
+
+// Example:
+// <Input name="email" type="email" label="Email Address"
+// required={true} validation="/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/" />
+
+class Input extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      inputText: "",
+      validated: true
+    }
+    this.setInputText = this.setInputText.bind(this);
+    this.checkValidation = this.checkValidation.bind(this);
```

Comment:
does CRA not have class properties out of the box? You shouldnt need to do this if you define your class methods like:

```
class Comp extends Component {
  someMethod = () => {
    ...
  }
}
```

Reply:
The fat arrow method is a lot cleaner.  Perhaps you guys should review the code of the instructors at Make School.  :)

Reply:
:P

I should also point out that you can define the state the same way. Shouldn't need to define the constructor fn at all